onnxruntime/contrib_ops/cuda/bert/attention_softmax.cu

```diff
@@ -172,7 +165,7 @@ __device__ inline void SoftmaxSmall(const int all_sequence_length,
 
   // threadIdx.x might be larger than all_sequence_length due to alignment to 32x.
   if (threadIdx.x < all_sequence_length) {
-    output[index] = T(thread_data_exp * sum_reverse_block);
+    output[index] = is_valid ? T(thread_data_exp * sum_reverse_block) : T(0.f);
```

Comment:
Add a check so that the code is more robust:  when all words are masked in a row, this will set masked value to 0.0 instead of NaN (caused by divided by zero of sum).